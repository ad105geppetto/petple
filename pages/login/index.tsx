import Signup from "../../src/components/units/signup/Signup.container";

export default function LoginPage() {
  return <Signup isSignup={false} />;
}
