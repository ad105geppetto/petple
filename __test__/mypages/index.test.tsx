import "@testing-library/jest-dom";
import { MockedProvider } from "@apollo/client/testing";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { mocks, propsMock } from "./login.mocks";
import { useRouter } from "next/router";
import { RecoilRoot } from "recoil";
import MypageUI from "../../src/components/units/mypage/Mypage.presenter";
import Mypage from "../../src/components/units/mypage/Mypage.container";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("마이페이지 Unit Test & Integration Test", () => {
  it("마이페이지 스냅샷 테스트", () => {
    const result = render(
      <RecoilRoot>
        <MockedProvider mocks={mocks} addTypename={false}>
          <MypageUI {...propsMock} />
        </MockedProvider>
      </RecoilRoot>
    );

    expect(result.container).toMatchSnapshot();
  });

  it("필수 정보 입력 후 비밀번호 변경 버튼 클릭시, 페이지 이동합니다.", async () => {
    const router = {
      push: jest.fn(),
    };
    (useRouter as jest.Mock).mockReturnValue(router);

    render(
      <RecoilRoot>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Mypage />
        </MockedProvider>
      </RecoilRoot>
    );

    const password = screen.getByPlaceholderText("새 비밀번호를 입력해주세요.");
    fireEvent.change(password, { target: { value: "qwerty123" } });
    //prettier-ignore
    const passwordCheck = screen.getByPlaceholderText("새 비밀번호를 확인해주세요.");
    fireEvent.change(passwordCheck, { target: { value: "qwerty123" } });

    const onClickLogin = screen.getByText("비밀번호 변경");
    fireEvent.click(onClickLogin);

    await waitFor(() => {
      const onClickOK = screen.getByText("OK");
      fireEvent.click(onClickOK);
    });

    expect(router.push).toHaveBeenCalledWith("/");
  });
});
