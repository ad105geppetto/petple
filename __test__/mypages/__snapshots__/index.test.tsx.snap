// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`마이페이지 Unit Test & Integration Test 마이페이지 스냅샷 테스트 1`] = `
<div>
  
  
  <h1
    class="css-17p1ae5"
  >
    개인정보 
    &gt;
     비밀번호 변경
  </h1>
  <div
    class="css-1lodiei"
  >
    새 비밀번호
    <input
      autocomplete="off"
      class="css-198g93x"
      placeholder="새 비밀번호를 입력해주세요."
      type="password"
    />
  </div>
  <div
    class="css-1lodiei"
  >
    새 비밀번호 확인
    <input
      autocomplete="off"
      class="css-198g93x"
      placeholder="새 비밀번호를 확인해주세요."
      type="password"
    />
  </div>
  <button
    class="css-11fm0ms"
  >
    비밀번호 변경
  </button>
</div>
`;
